packages/map/src/map/hooks.jsx

```diff
@@ -1,65 +1,69 @@
-import { useEffect, useState, useRef, useContext } from 'react';
-import ReactMapboxGl from 'react-mapbox-gl';
+import { useEffect, useState, useContext, useMemo } from 'react';
+import { Map, NavigationControl } from 'mapbox-gl';
+
+import { getMapOptions, getBoundingBox, mergeChildrenBounds } from './utils';
 import { getMedia, media } from '../utils';
 import MapContext from './context';
 
 
-export const useLocked = (locked, lockedMobile) => {
-  const [isLocked, setLocked] = useState(locked);
+export const useMapContext = () => useContext(MapContext);
+
+export const useMapInstance = (nodeRef, options) => {
+  const [mapInstance, setMap] = useState(false);
+  const { noAttribution, locked, lockedMobile, onLoad } = options;
 
   useEffect(() => {
-    const isMobile = !getMedia(global, media.mediaM);
-    setLocked(isMobile ? lockedMobile : locked);
-  }, [locked, lockedMobile]);
+    const mapOptions = getMapOptions({
+      ...options,
+      node: nodeRef.current,
+      isMobile: !getMedia(global, media.mediaM),
+    });
 
-  return isLocked;
-};
+    const map = new Map(mapOptions);
 
-export const useMapboxComponent = (isLocked, noAttribution) => {
-  const [, setState] = useState();
-  const ref = useRef(null);
+    const handleLoad = () => {
+      if (onLoad) onLoad();
+      // Need to wait for style load before rendering layers
+      setMap(map);
+    };
 
-  useEffect(() => {
-    ref.current = ReactMapboxGl({
-      interactive: !isLocked,
-      keyboard: false,
-      doubleClickZoom: false,
-      scrollZoom: false,
-      dragRotate: false,
-      pitchWithRotate: false,
-      apiUrl: null,
-      injectCSS: false,
-      attributionControl: !noAttribution,
-    });
+    map.once('load', handleLoad);
+    if (mapOptions.interactive) map.addControl(new NavigationControl());
 
-    // Force re-render
-    setState({});
-  }, [isLocked, noAttribution]);
+    return () => {
+      setMap(null);
+      if (map) map.remove();
+    };
+  }, [noAttribution, locked, lockedMobile]);
 
-  return ref.current;
+  return mapInstance;
 };
 
-export const useDefaultCenterAndZoom = (defaultZoom, defaultView) => {
-  const zoomRef = useRef(defaultZoom ? [defaultZoom] : undefined);
-  const centerRef = useRef(defaultView || undefined);
+export const useMapUpdate = (map, { bounds, childrenBounds, animate, fitPadding, defaultView }) => {
+  const boundsToFit = bounds || mergeChildrenBounds(childrenBounds);
+  const boundsSignature = JSON.stringify(boundsToFit);
+
+  useEffect(() => {
+    const boundingBox = getBoundingBox(boundsToFit);
 
-  return [centerRef.current, zoomRef.current];
+    if (map && boundingBox && !defaultView) {
+      map.fitBounds(boundingBox, { animate, padding: fitPadding });
+    }
+  }, [boundsSignature, map]);
 };
 
-export const useContextValue = () => {
+export const useContextValue = (map) => {
   const [bounds, setBounds] = useState([]);
+  const addBounds = (value) => {
```

Comment:
use `useCallback` to prevent unneeded renders ? (context will cause a re-render because `addBounds` is always a different object)

Reply:
no need. it's then used in useMemo